leakcanary-android-core/src/main/AndroidManifest.xml

```diff
@@ -43,7 +43,28 @@
         android:label="@string/leak_canary_display_activity_label"
         android:taskAffinity="com.squareup.leakcanary.${applicationId}"
         android:theme="@style/leak_canary_LeakCanary.Base"
-        />
+        >
+      <intent-filter android:label="@string/leak_canary_import_hprof_file">
+        <action android:name="android.intent.action.VIEW"/>
+
+        <category android:name="android.intent.category.DEFAULT"/>
+        <category android:name="android.intent.category.BROWSABLE"/>
+
+        <data android:scheme="file"/>
+        <data android:scheme="content"/>
+        <data android:mimeType="*/*"/>
+        <data android:host="*"/>
+
+        <data android:pathPattern=".*\\.hprof"/>
+        <data android:pathPattern=".*\\..*\\.hprof"/>
+        <data android:pathPattern=".*\\..*\\..*\\.hprof"/>
+        <data android:pathPattern=".*\\..*\\..*\\..*\\.hprof"/>
+        <data android:pathPattern=".*\\..*\\..*\\..*\\..*\\.hprof"/>
+        <data android:pathPattern=".*\\..*\\..*\\..*\\..*\\..*\\.hprof"/>
+        <data android:pathPattern=".*\\..*\\..*\\..*\\..*\\..*\\..*\\.hprof"/>
```

Comment:
Fascinating. What made you stop at a depth of 7?

Reply:
Magic number 🤓 
There is no real reasoning here.
It really relies on apps to not insert too much `.` separators on their Uri.
I'd hope `pathPattern` would have better support for regex but it's quite limited IIRC.

Reply:
Also you might want to mention the source, e.g. is this from https://stackoverflow.com/a/31028507/703646 ?

In any case it'd be great to have a bit more clarifying comments directly there for why this is there.

Reply:
oooh I see. So that's because segments of the path contain periods because they're packages. Great. So there's no max but generally don't expect too many I guess.

Reply:
I'll add an inline comment with the source and the reason then.